calci.c

```diff
@@ -0,0 +1,190 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <stdbool.h>
+
+#define MAX 100
+
+
+typedef struct {
+    int data[MAX];
+    int top;
+} IntStack;
+
+
+typedef struct {
+    char data[MAX];
+    int top;
+} OpStack;
+
+
+void initIntStack(IntStack* stack) {
+    stack->top = -1;
+}
+
+
+void pushInt(IntStack* stack, int value) {
+    if (stack->top >= MAX - 1) {
+        printf("Error: Stack overflow.\n");
+        exit(1);
+    }
+    stack->data[++stack->top] = value;
+}
+
+
+int popInt(IntStack* stack) {
+    if (stack->top < 0) {
+        printf("Error: Invalid Expression.\n");
+        exit(1);
+    }
+    return stack->data[stack->top--];
+}
+
+
+int peekInt(IntStack* stack) {
+    if (stack->top < 0) {
+        printf("Error: Stack is empty.\n");
+        exit(1);
+    }
+    return stack->data[stack->top];
+}
+
+
+void initOpStack(OpStack* stack) {
+    stack->top = -1;
+}
+
+
+void pushChar(OpStack* stack, char value) {
+    if (stack->top >= MAX - 1) {
+        printf("Error: Stack overflow.\n");
+        exit(1);
+    }
+    stack->data[++stack->top] = value;
+}
+
+
+char popChar(OpStack* stack) {
+    if (stack->top < 0) {
+        printf("Error: Invalid Expression.\n");
+        exit(1);
+    }
+    return stack->data[stack->top--];
+}
+
+
+char peekChar(OpStack* stack) {
+    if (stack->top < 0) {
+        printf("Error: Invalid Expression.\n");
+        exit(1);
+    }
+    return stack->data[stack->top];
+}
+
+
+bool isDigit(char c) {
+    return c >= '0' && c <= '9';
+}
+
+
+bool isSpace(char c) {
+    return c == ' ' || c == '\t' || c == '\n';
+}
+
+
+int precedence(char op) {
+    if (op == '+' || op == '-') return 1;
+    if (op == '*' || op == '/') return 2;
+    return 0;
+}
+
+
+int applyOperation(int a, int b, char op) {
+    switch (op) {
+        case '+': return a + b;
+        case '-': return a - b;
+        case '*': return a * b;
+        case '/':
+            if (b == 0) {
+                printf("Error: Division by zero.\n");
+                exit(1);
+            }
+            return a / b;
+        default:
+            printf("Error: Invalid Expression.\n");
+            exit(1);
+    }
+}
+
+int stringLength(const char* str){
+    int i;
+    for(i=0; str[i] != '\0'; i++){}

```

Comment:
You can use while loop here, which makes more sense instead of running empty for loop.

Reply:
Okay, I have changed the empty for loop with a while loop.